### PR TITLE
MUI select conversion

### PR DESCRIPTION
**Pull request by agalin920**
Screen Shot 2022-06-27 at 5 56 13 PM
Screen Shot 2022-06-27 at 5 56 59 PM
Screen Shot 2022-06-27 at 6 26 35 PM
Screen Shot 2022-06-27 at 6 26 41 PM
Screen Shot 2022-06-27 at 6 55 03 PM
Screen Shot 2022-06-27 at 6 55 07 PM
Screen Shot 2022-06-27 at 7 00 56 PM
Screen Shot 2022-06-27 at 7 01 01 PM
Screen Shot 2022-06-27 at 7 01 40 PM
Screen Shot 2022-06-27 at 7 32 26 PM
Screen Shot 2022-06-27 at 7 32 50 PM
Screen Shot 2022-06-27 at 8 16 52 PM
Screen Shot 2022-06-27 at 8 16 58 PM
Screen Shot 2022-06-27 at 8 17 08 PM
Screen Shot 2022-06-27 at 8 19 05 PM
Screen Shot 2022-06-27 at 8 26 15 PM
Screen Shot 2022-06-27 at 8 26 20 PM
Screen Shot 2022-06-27 at 8 26 27 PM
Screen Shot 2022-06-27 at 8 26 32 PM
Screen Shot 2022-06-27 at 8 36 31 PM
Screen Shot 2022-06-27 at 8 36 35 PM
Screen Shot 2022-06-27 at 8 36 41 PM
Screen Shot 2022-06-27 at 8 44 59 PM
Screen Shot 2022-06-27 at 8 55 58 PM
Screen Shot 2022-06-27 at 9 06 59 PM
Screen Shot 2022-06-27 at 9 09 57 PM
Screen Shot 2022-06-27 at 9 15 55 PM
Screen Shot 2022-06-27 at 9 16 04 PM
Screen Shot 2022-06-27 at 9 36 55 PM
Screen Shot 2022-06-27 at 9 50 34 PM
Screen Shot 2022-06-27 at 9 50 37 PM
Screen Shot 2022-06-27 at 9 50 40 PM
Screen Shot 2022-06-27 at 9 58 53 PM
Screen Shot 2022-06-27 at 10 19 46 PM

### CODE REVIEW

File: cypress/e2e/content/navigation.spec.js [hunk 1]
```diff
@@ -20,11 +20,16 @@ describe("Navigation through content editor", () => {
     cy.get("#MainNavigation").then((content) => {
       if (!content.is(":visible")) {
         cy.get("[data-cy=contentNavButton]").click();
-        cy.get(".CreateItemDropdown").click(); // IF another user modal is up force click
-        cy.get('[data-value="link"]').click();
+        cy.get(".CreateItemDropdown")
+          .find(".MuiSelect-select")
+          .click({ force: true }); // IF another user modal is up force click
```
Comment (shrunyan) [inline cypress/e2e/content/navigation.spec.js:25]: we should no longer need force clicks

File: cypress/e2e/content/navigation.spec.js [hunk 2]
```diff
@@ -20,11 +20,16 @@ describe("Navigation through content editor", () => {
...
+        cy.get("[role=presentation]").find('[data-value="link"]').click();
         cy.get("#CreateLinkButton").should("exist");
       } else {
-        cy.get(".CreateItemDropdown").click(); // IF another user modal is up force click
+        cy.get(".CreateItemDropdown")
+          .find(".MuiSelect-select")
+          .click({ force: true }); // IF another user modal is up force click
```
Comment (shrunyan) [inline cypress/e2e/content/navigation.spec.js:31]: we should no longer need force clicks

File: cypress/e2e/releases/members.spec.js [hunk 3]
```diff
@@ -62,14 +64,11 @@ describe("Release > members > CRUD", () => {
         pathname: "/v1/releases/27-d0d8f7a0f8-1pp779/members/*",
       },
       () => {
-        cy.get(
-          "[data-cy=PlanTable] tbody tr:last-child [data-cy=release-member-version] .Select .options li:last-child"
-        ).click();
+        cy.get("[role=presentation] li:last-child").click();
```
Comment (shrunyan) [inline cypress/e2e/releases/members.spec.js:67]: why is this being changed to a less specific selector?

File: cypress/e2e/releases/members.spec.js [hunk 4]
```diff
@@ -44,15 +44,17 @@ describe("Release > members > CRUD", () => {
...
     });
+    cy.get("#root").click();
 
     cy.get("[data-cy=PlanTable] tbody tr:last-child")
       .contains(`Homepage`)
       .should("exist");
   });
 
   it("update member", () => {
+    cy.wait(5000);
```
Comment (shrunyan) [inline cypress/e2e/releases/members.spec.js:55]: why is there a wait being added?

File: src/apps/active-preview/Preview.js [hunk 5]
```diff
@@ -191,23 +188,21 @@ export function Preview(props) {
               <Select
                 className={styles.Select}
                 name="device"
-                value="fullscreen"
-                onSelect={(val) => setDevice(val)}
+                value={device}
+                onChange={(e) => setDevice(e.target.value)}
```
Comment (shrunyan) [inline src/apps/active-preview/Preview.js:192]: lets avoid single character variables.I would prefer this to be `evt` to express that it is an event

File: src/apps/code-editor/src/app/components/FileActions/components/DifferActions/DifferActions.js [hunk 6]
```diff
@@ -141,9 +149,13 @@ export const DifferActions = memo(function DifferActions(props) {
...
-        value={options[0] && options[0].value}
-        onSelect={(value) => {
-          const version = versions.find((version) => version.version == value);
+        defaultValue="local"
+        size="small"
+        onChange={(e) => {
+          const version = versions.find(
+            (version) => version.version == e.target.value
+          );
+          console.log("testing change", version);
```
Comment (shrunyan) [inline src/apps/code-editor/src/app/components/FileActions/components/DifferActions/DifferActions.js:158]: remove log

File: src/apps/code-editor/src/app/components/FileActions/components/DifferActions/DifferActions.js [hunk 7]
```diff
@@ -141,9 +149,13 @@ export const DifferActions = memo(function DifferActions(props) {
...
         name="codeOne"
         className={styles.VersionSelector}
-        value={options[0] && options[0].value}
-        onSelect={(value) => {
-          const version = versions.find((version) => version.version == value);
+        defaultValue="local"
+        size="small"
+        onChange={(e) => {
+          const version = versions.find(
+            (version) => version.version == e.target.value
```
Comment (shrunyan) [inline src/apps/code-editor/src/app/components/FileActions/components/DifferActions/DifferActions.js:156]: avoid single character variables

File: src/apps/code-editor/src/app/components/FileActions/components/DifferActions/DifferActions.js [hunk 8]
```diff
@@ -162,8 +176,11 @@ export const DifferActions = memo(function DifferActions(props) {
         name="codeTwo"
         className={styles.VersionSelector}
         value={selectedVersion}
-        onSelect={(value) => {
-          const version = versions.find((version) => version.version == value);
+        size="small"
+        onChange={(e) => {
```
Comment (shrunyan) [inline src/apps/code-editor/src/app/components/FileActions/components/DifferActions/DifferActions.js:180]: avoid single character variables

File: src/apps/content-editor/src/app/components/Nav/ContentNav.js [hunk 9]
```diff
@@ -78,7 +77,8 @@ export function ContentNav(props) {
     dispatch(collapseNavItem(nodeProps.path));
   };
 
-  const handleCreateSelect = (ZUID) => {
+  const handleCreateSelect = (e) => {
```
Comment (shrunyan) [inline src/apps/content-editor/src/app/components/Nav/ContentNav.js:80]: avoid single character variables

File: src/apps/content-editor/src/app/views/CSVImport/Columns/Columns.js [hunk 10]
```diff
@@ -22,18 +22,17 @@ export default function Columns(props) {
               <span className={cx(styles.Cell)}>{col.toUpperCase()}</span>
               <Select
                 name={colName}
-                onSelect={(value) => {
-                  props.handleMap(value, col);
+                onChange={(e) => {
```
Comment (shrunyan) [inline src/apps/content-editor/src/app/views/CSVImport/Columns/Columns.js:25]: avoid single character variables

File: src/apps/content-editor/src/app/views/CSVImport/CsvSettings/CsvSettings.js [hunk 11]
```diff
@@ -10,57 +10,101 @@ export const CsvSettings = (props) => {
...
           ))}
         </Select>
       </article>
       <article className={styles.Setting}>
         <label>Meta Keywords</label>
-        <Select name="metaKeywords" onSelect={props.handleMap} value="none">
-          <Option text="none" value="none" />
+        <Select
+          name="metaKeywords"
+          onChange={(e) => props.handleMap(e.target.value, "metaKeywords")}
```
Comment (shrunyan) [inline src/apps/content-editor/src/app/views/CSVImport/CsvSettings/CsvSettings.js:30]: avoid single character variables

File: src/apps/content-editor/src/app/views/ItemEdit/components/Header/LanguageSelector/LanguageSelector.js [hunk 12]
```diff
@@ -29,7 +29,8 @@ export const LanguageSelector = connect((state, props) => {
   const location = useLocation();
   const history = useHistory();
 
-  const handleSelect = (val) => {
+  const handleSelect = (e) => {
```
Comment (shrunyan) [inline src/apps/content-editor/src/app/views/ItemEdit/components/Header/LanguageSelector/LanguageSelector.js:32]: avoid single character variables

File: src/apps/content-editor/src/app/views/LinkCreate/LinkCreate.js [hunk 13]
```diff
@@ -135,29 +139,23 @@ export function LinkCreate() {
             <>
               {" "}
               <Select
-                label="Select link type"
                 name="type"
                 value={state.type}
-                onSelect={onChange}
+                onChange={(e) => onChange(e.target.value, "type")}
```
Comment (shrunyan) [inline src/apps/content-editor/src/app/views/LinkCreate/LinkCreate.js:144]: avoid single character variables

File: src/apps/media/src/app/components/MediaDetailsModal.js [hunk 14]
```diff
@@ -162,17 +158,26 @@ export const MediaDetailsModal = memo(function MediaDetailsModal(props) {
                       className={styles.Select}
                       name="optimize"
                       value={imageSettings.optimize}
-                      onSelect={(value) =>
+                      onChange={(e) =>
```
Comment (shrunyan) [inline src/apps/media/src/app/components/MediaDetailsModal.js:161]: avoid single character variables

File: src/apps/media/src/app/components/MediaDetailsModal.js [hunk 15]
```diff
@@ -181,17 +186,26 @@ export const MediaDetailsModal = memo(function MediaDetailsModal(props) {
                       className={styles.Select}
                       name="fit"
                       value={imageSettings.fit}
-                      onSelect={(value) =>
+                      onChange={(e) =>
```
Comment (shrunyan) [inline src/apps/media/src/app/components/MediaDetailsModal.js:189]: avoid single character variables

File: src/apps/release/src/app/views/ViewRelease/components/Header/Header.js [hunk 16]
```diff
@@ -47,15 +46,14 @@ export function Header({ plan }) {
       <Select
         name="release"
         value={params.zuid}
-        onSelect={(val) => history.push(`/release/${val}`)}
+        onChange={(e) => history.push(`/release/${e.target.value}`)}
```
Comment (shrunyan) [inline src/apps/release/src/app/views/ViewRelease/components/Header/Header.js:49]: avoid single character variables

File: src/apps/release/src/app/views/ViewRelease/components/PlanStep/PlanStep.js [hunk 17]
```diff
@@ -124,12 +127,16 @@ export function PlanStep(props) {
       <td data-cy="release-member-version">
         {/* Update preview link when version is changed */}
         <Select
-          onSelect={onUpdateVersion}
+          onChange={(e) => onUpdateVersion(e.target.value)}
```
Comment (shrunyan) [inline src/apps/release/src/app/views/ViewRelease/components/PlanStep/PlanStep.js:130]: avoid single character variables

File: src/apps/seo/src/views/RedirectsManager/RedirectImportTable/RedirectImportTableRow/RedirectImportTableRow.js [hunk 18]
```diff
@@ -70,21 +72,26 @@ function RedirectImportTableRow(props) {
 
       <span className={styles.RowCell}>
         {props.target_type === "page" ? (
-          <Select className={styles.selector} onSelect={handlePageTarget}>
+          <Select
+            className={styles.selector}
+            onChange={(e) => handlePageTarget(e.target.value)}
```
Comment (shrunyan) [inline src/apps/seo/src/views/RedirectsManager/RedirectImportTable/RedirectImportTableRow/RedirectImportTableRow.js:77]: avoid single character variables

File: src/apps/seo/src/views/RedirectsManager/RedirectsTable/RedirectCreator/RedirectCreator.js [hunk 19]
```diff
@@ -86,10 +87,22 @@ export function RedirectCreator(props) {
         </ToggleButtonGroup>
       </span>
       <span className={styles.RedirectCreatorCell}>
-        <Select name={"selectType"} onSelect={setType} value={type}>
-          <Option value="page" selected text="Internal" />
-          <Option value="external" text="External" />
-          <Option value="path" text="Wildcard" />
+        <Select
+          name={"selectType"}
+          onChange={(e) => setType(e.target.value)}
```
Comment (shrunyan) [inline src/apps/seo/src/views/RedirectsManager/RedirectsTable/RedirectCreator/RedirectCreator.js:92]: avoid single character variables

File: src/apps/settings/src/app/views/Instance/Instance.js [hunk 20]
```diff
@@ -181,17 +184,15 @@ export default connect((state) => {
                     <div className={styles.selectProtocol}>
                       <Select
                         name={field.key}
-                        onSelect={setValue}
+                        onChange={(e) => setValue(e.target.value, field.key)}
```
Comment (shrunyan) [inline src/apps/settings/src/app/views/Instance/Instance.js:187]: avoid single character variables

File: src/apps/settings/src/app/views/Instance/Instance.js [hunk 21]
```diff
@@ -295,18 +296,16 @@ export default connect((state) => {
                   <div className={styles.selectProtocol}>
                     <Select
                       name={field.key}
-                      onSelect={setValue}
+                      onChange={(e) => setValue(e.target.value, field.key)}
```
Comment (shrunyan) [inline src/apps/settings/src/app/views/Instance/Instance.js:299]: avoid single character variables

File: src/apps/settings/src/app/views/Styles/Styles.js [hunk 22]
```diff
@@ -207,27 +205,30 @@ export default connect((state) => {
             <span style={{ fontSize: "1.5rem" }}>{field.keyFriendly}</span>
             <div className={styles.fontPicker}>
               <Select
+                fullWidth
                 name={field.referenceName}
-                onSelect={setValue}
+                onChange={(e) => setValue(e.target.value, field.referenceName)}
```
Comment (shrunyan) [inline src/apps/settings/src/app/views/Styles/Styles.js:210]: avoid single character variables

File: cypress/e2e/releases/members.spec.js (hunk 4 above)
Comment (agalin920) [reply cypress/e2e/releases/members.spec.js:55]: Accidently leftover. Removing

File: cypress/e2e/releases/members.spec.js (hunk 3 above)
Comment (agalin920) [reply cypress/e2e/releases/members.spec.js:67]: Because the popover is rendered on a portal it no longer follows that dom tree selector

**Review by shrunyan — APPROVED** [review]

**Pull request merged**